tools/precommit-clang-format.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
```

Comment:
Can this run on windows?

Reply:
or on a system without `bash`

Reply:
Great point, I've only tested on my Windows install that has `sh`. Maybe it would be good to port this to a Python script instead, I can do that easy

Reply:
Alright, ported this to a python script at da90a7652c0d4a5ed2690af7460094268d9c4a86